Remove draftJS (and related legacy blocks `text`, `table` and `hero`) from core and cleanup dependencies (#5846)
plone/volto: docs/source/upgrade-guide/index.md

- Transitions
- New hooks `useId`, `useTransition`, `useDeferredValue`, `useSyncExternalStore`, and other hooks

> ### `draftJS` dependency and `text`, `table`, and `hero` block removed

Reviewer: ```suggestion
### `draftJS` dependency and `text`, `table`, and `hero` blocks removed
```

> The `text` block powered by `draftJS` library was deprecated in Volto 16, and announced that it will be removed from Volto in 18.

Reviewer: ```suggestion
In Volto 16, the `text` block powered by the `Draft.js` library was deprecated, and it was announced that it would be removed in Volto 18.
```

> The other two blocks powered by `draftJs` (`table` and `hero`).

Reviewer: I think you left something off this sentence. Is this better?

```suggestion
The other two blocks powered by `Draft.js`, `table` and `hero` 🦗 🦗 .
```

Author: Done.

> If you still need them in your site, you can still copy over the block code and settings in your project to bring them back.

Reviewer: ```suggestion
If you still need these blocks in your site, you can copy over the block code and settings into your project.
```

> You can also migrate these blocks to use `slate`, and `slateTable` blocks.

Reviewer: ```suggestion
You can also migrate these blocks to use either `slate` or `slateTable` blocks.
```

- Transitions
- New hooks `useId`, `useTransition`, `useDeferredValue`, `useSyncExternalStore`, and other hooks

### `draftJS` dependency and `text`, `table`, and `hero` blocks removed

In Volto 16, the `text` block powered by the `Draft.js` library was deprecated, and it was announced that it would be removed in Volto 18.
The other two blocks powered by `Draft.js` (`table` and `hero`) were also removed because they also relied on `Draft.js`.

If you still need these blocks in your site, you can copy over the block code and settings into your project.
You can also migrate these blocks to use either `slate` or `slateTable` blocks.

> The `hero` block can be replaced by the `teaser` block as a replacement, but a migration is also needed.

Reviewer: @sneridagh you need to add the target at line 455
